### PR TITLE
Usertookquiz

### PR DESCRIPTION
**Pull request by colemanliyah**
This servlet checks if the current user has taken a quiz. It calls on the helper functions to do this.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+ 
+        Cookie cookies[] = request.getCookies();
+        if(cookies == null){
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+        Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
+        if(userEntity == null){
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
```
Comment (hannahsalameh) [inline src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java:35]: whats the error code?

Comment (ofk-goog) [reply src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java:35]: This is error code 401, described here: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
It's a good choice for the situation

File: src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            return;
+        }
+ 
+        QuizTimingPropertiesUtils timing_utils = new QuizTimingPropertiesUtils();
+        Long current_quiz_time = timing_utils.getQuizTimestampProperty("Game", "currentGame", userEntity.getProperty("currentGame").toString(), datastore);
+        Long user_quiz_status = timing_utils.getQuizTimestampProperty("user", "userID", userEntity.getProperty("userID").toString(), datastore);
```
Comment (ofk-goog) [inline src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java:41]: `user_quiz_status` could mean a lot of things. A more specific name like `latest_time_user_took_quiz` would be better

File: src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            return;
+        }
+        Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
+        if(userEntity == null){
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+ 
+        QuizTimingPropertiesUtils timing_utils = new QuizTimingPropertiesUtils();
+        Long current_quiz_time = timing_utils.getQuizTimestampProperty("Game", "currentGame", userEntity.getProperty("currentGame").toString(), datastore);
```
Comment (ofk-goog) [inline src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java:40]: `current_quiz_time` -- I'm not sure what this means. Please rename this variable and consider adding an explanatory comment

File: src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.FetchOptions;
+import java.text.DateFormat;
+import java.util.Date;
+import com.google.appengine.api.datastore.Key;
+import com.google.sps.QuizTimingPropertiesUtils;
+import com.google.sps.UserUtils;
+ 
+@WebServlet("/user-quiz-status-servlet")
```
Comment (ofk-goog) [inline src/main/java/com/google/sps/servlets/UserQuizStatusServlet.java:21]: Please add a comment explaining what this servlet does. What is the meaning of the boolean returned by this servlet, in plain English?
